astropy/io/votable/validator/result.py

```diff
@@ -39,29 +38,29 @@ def __enter__(self):
     def __exit__(self, *args):
         self.save_attributes()
 
-    def get_dirpath(self):
-        return os.path.join(self._root, self._path)
+    def get_dirpath(self) -> Path:
+        return self._root / self._path
 
-    def get_htmlpath(self):
+    def get_htmlpath(self) -> Path:
         return self._path
 
-    def get_attribute_path(self):
-        return os.path.join(self.get_dirpath(), "values.dat")
+    def get_attribute_path(self) -> Path:
```

Comment:
I'd rather not. This is API change and I use this downstream in `astroquery`.

Reply:
In general, we should probably leave validator alone.

Reply:
I agree this should be done in a separate PR as an API change. But I think this is a good thing to do long term. We should probably also consider deprecating getter and setter methods in favor of `@property`.

Reply:
Oh I don't know why I thought this would be private API. I'll revert the breaking change for now, but it should be simple enough to futureproof astroquery against such a change: wrapping all calls to this methods with `os.fspath()` would do the trick.

Reply:
I revamped this as a set of `Path` returning private properties + `str` returning public accessors: no API breakage yet, but a clear path forward: properties can be made public and accessors deprecated. What do you guys think ?